imports/server/guesses.ts

```diff
@@ -74,17 +72,15 @@ class PendingGuessWatcher {
 
   puzzleRefCounter: RefCountedObserverMap<PuzzleType>;
 
-  constructor(sub: Subscription) {
+  constructor(sub: Subscription, huntId: string) {
     this.sub = sub;
 
-    const user = MeteorUsers.findOne(sub.userId!)!;
-
-    this.guessCursor = Guesses.find({ state: 'pending', hunt: { $in: user.hunts } });
+    const guessCursor = Guesses.find({ state: 'pending', hunt: huntId });
     this.guesses = {};
     this.huntRefCounter = new RefCountedObserverMap(sub, Hunts);
```

Comment:
I was going to comment that it's generally better to try to share these by passing them into the constructor so that you don't double-call e.g. `added()` with the same object multiple times on the same sub, but in practice the Hunts and Puzzles for each one of these `HuntPendingGuessWatcher`s will be orthogonal, so it doesn't matter in this case.

If you were doing something where data didn't necessarily fall completely orthogonally in the same way, like reactively publishing display names for users that might overlap between hunts, then you would want to reuse the same RefCountedObserverMap for each `HuntPendingGuessWatcher`.  Which...I guess we always subscribe to all display names everywhere?  In the fullness of time maybe we'll want to make that more granular or scoped.  For now, we need the display names on each page we might show the pending guess queue, which is all pages.  We don't have to change that now, but it feels like a potential win on the fetching-data-as-needed side -- if I remember correctly, in prod I think a substantial fraction of our initial data loading is profiles/display names.

Reply:
Yeah we could be handling display names a lot better than we do now.